### PR TITLE
[Qumulo.Storage] APEX - Create a new module: Qumulo, and support it auto generated

### PR DESCRIPTION
**Pull request by JoyerJin**
## Description

Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/liftrqumulo/resource-manager/Qumulo.Storage/stable/2022-10-12/Qumulo.Storage.json
Design: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1338

* Update cmdlet only have tags parameter
* Set 'AdminPassword' as a secret string
* Rename parameters
* Rename resource description

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Qumulo/examples/Get-AzQumuloFileSystem.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+eastus   qumulo-01          6/24/2023 5:27:12 AM user@organization.com User                    6/24/2023 5:27:12 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-test      
+eastus   qumulo-02          6/24/2023 5:31:50 AM user@organization.com User                    6/24/2023 5:31:50 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-test
+eastus   fileSystem         5/24/2023 7:10:01 AM user@organization.com User                    5/24/2023 7:19:16 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-01 5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 7:42:17 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-02 5/24/2023 9:31:50 AM user@organization.com User                    5/24/2023 9:41:10 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test
+```
+
+Get list of File System Resource by subscription
```
Comment (vidai-msft) [inline src/Qumulo/examples/Get-AzQumuloFileSystem.md:17]:
```suggestion
Get list of file system resources by subscription
```

File: src/Qumulo/examples/Get-AzQumuloFileSystem.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+```output
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy             SystemDataLastModifiedByType ResourceGroupName  
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------             ---------------------------- -----------------  
+eastus   fileSystem         5/24/2023 7:10:01 AM user@organization.com User                    5/24/2023 7:19:16 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-01 5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 7:42:17 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-02 5/24/2023 9:31:50 AM user@organization.com User                    5/24/2023 9:41:10 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test 
+```
+
+Get list of File System Resource by resource group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Get-AzQumuloFileSystem.md:32]:
```suggestion
Get list of file system resources by resource group
```

File: src/Qumulo/examples/Get-AzQumuloFileSystem.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+Get-AzQumuloFileSystem -ResourceGroupName azpstest-gp -Name fileSystem
+```
+
+```output
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy SystemDataLastModifiedByType ResourceGroupName
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------ ---------------------------- -----------------
+eastus   qumulo-resource-01 5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 9:58:45 AM     user@organization.com    User                         ps-joyer-test 
+```
+
+Get special File System Resource with special resource group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Get-AzQumuloFileSystem.md:45]:
```suggestion
Get specific file system resource with specified resource group
```

File: src/Qumulo/examples/New-AzQumuloFileSystem.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+```output
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy             SystemDataLastModifiedByType ResourceGroupName  
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------             ---------------------------- -----------------
+eastus   qumulo01           5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 7:42:17 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test
+```
+
+Create a FileSystemResource. The password must contain at least 8 characters and have at least 1 letter, 1 number and 1 special character.
```
Comment (vidai-msft) [inline src/Qumulo/examples/New-AzQumuloFileSystem.md:14]:
```suggestion
Create a file system resource. The password must contain at least 8 characters and have at least 1 letter, 1 number and 1 special character
```

File: src/Qumulo/examples/New-AzQumuloFileSystem.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+```output
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy             SystemDataLastModifiedByType ResourceGroupName  
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------             ---------------------------- -----------------  
+eastus   qumulo02           5/24/2023 9:31:50 AM user@organization.com User                    5/24/2023 9:41:10 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test
+```
+
+Create a FileSystemResource with a maximum set
```
Comment (vidai-msft) [inline src/Qumulo/examples/New-AzQumuloFileSystem.md:29]:
```suggestion
Create a file system resource with a maximum set
```

File: src/Qumulo/examples/Remove-AzQumuloFileSystem.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+### Example 1: Remove special resource with sepecial group
+```powershell
+Remove-AzQumuloFileSystem -Name qumulo01 -ResourceGroupName ps-joyer-test02
+```
+
+Remove special File System Resource in sepecial group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Remove-AzQumuloFileSystem.md:6]:
```suggestion
Remove specific file system resource in specified resource group
```

File: src/Qumulo/examples/Get-AzQumuloFileSystem.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy             SystemDataLastModifiedByType ResourceGroupName  
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------             ---------------------------- -----------------  
+eastus   fileSystem         5/24/2023 7:10:01 AM user@organization.com User                    5/24/2023 7:19:16 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-01 5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 7:42:17 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test      
+eastus   qumulo-resource-02 5/24/2023 9:31:50 AM user@organization.com User                    5/24/2023 9:41:10 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test 
+```
+
+Get list of File System Resource by resource group
+
+### Example 3: Get special resource with sepecial group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Get-AzQumuloFileSystem.md:34]:
```suggestion
### Example 3: Get specific resource with specified resource group
```

File: src/Qumulo/examples/New-AzQumuloFileSystem.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+### Example 1: Create a minimum set FileSystemResource
```
Comment (vidai-msft) [inline src/Qumulo/examples/New-AzQumuloFileSystem.md:1]:
```suggestion
### Example 1: Create a minimum set file system resource
```

File: src/Qumulo/examples/New-AzQumuloFileSystem.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+```output
+Location Name               SystemDataCreatedAt  SystemDataCreatedBy   SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy             SystemDataLastModifiedByType ResourceGroupName  
+-------- ----               -------------------  -------------------   ----------------------- ------------------------ ------------------------             ---------------------------- -----------------
+eastus   qumulo01           5/24/2023 7:27:12 AM user@organization.com User                    5/24/2023 7:42:17 AM     13c34964-a135-4390-aa53-32f3c7251982 Application                  ps-joyer-test
+```
+
+Create a FileSystemResource. The password must contain at least 8 characters and have at least 1 letter, 1 number and 1 special character.
+
+### Example 2: Create a FileSystemResource with other settings
```
Comment (vidai-msft) [inline src/Qumulo/examples/New-AzQumuloFileSystem.md:16]:
```suggestion
### Example 2: Create a file system resource with other settings
```

File: src/Qumulo/examples/Remove-AzQumuloFileSystem.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
+### Example 1: Remove special resource with sepecial group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Remove-AzQumuloFileSystem.md:1]:
```suggestion
### Example 1: Remove specific resource with specified resource group
```

File: src/Qumulo/examples/Remove-AzQumuloFileSystem.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+```powershell
+Remove-AzQumuloFileSystem -Name qumulo01 -ResourceGroupName ps-joyer-test02
+```
+
+Remove special File System Resource in sepecial group
+
+### Example 2: Get and remove special resource with sepecial group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Remove-AzQumuloFileSystem.md:8]:
```suggestion
### Example 2: Get and remove specific resource with specified resource group
```

File: src/Qumulo/examples/Remove-AzQumuloFileSystem.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+```powershell
+$fileSystem = Get-AzQumuloFileSystem -ResourceGroupName ps-joyer-test -Name qumulo-resource-01
+Remove-AzQumuloFileSystem -InputObject $fileSystem
+```
+
+Remove special File System Resource in sepecial group
```
Comment (vidai-msft) [inline src/Qumulo/examples/Remove-AzQumuloFileSystem.md:14]:
```suggestion
Remove specific file system resource in specified resource group
```

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Comment by vidai-msft** [comment]
@JoyerJin Please also fix the issues detected in static analysis

File: src/Qumulo/docs/Update-AzQumuloFileSystem.md [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+external help file:
+Module Name: Az.Qumulo
+online version: https://learn.microsoft.com/powershell/module/az.qumulo/update-azqumulofilesystem
+schema: 2.0.0
+---
+
+# Update-AzQumuloFileSystem
+
+## SYNOPSIS
+Update a FileSystemResource
```
Comment (vidai-msft) [inline src/Qumulo/docs/Update-AzQumuloFileSystem.md:11]: Update does not take effect

**Review by vidai-msft — CHANGES_REQUESTED** [review]

File: src/Qumulo/docs/Update-AzQumuloFileSystem.md (hunk 13 above)
Comment (JoyerJin) [reply src/Qumulo/docs/Update-AzQumuloFileSystem.md:11]: Customed cmdlet will be covered, new knowledge.

**Review by vidai-msft — APPROVED** [review]

**Pull request merged**